### ISSUE TITLE
Disability Emojis aren't supported yet, but others are

### ISSUE THREAD

**Issue by alequetzalli** [thepracticaldev/dev.to#5835]
**Describe the bug**

Hello :) 

I want to use the service dog emoji in my dev to posts, but they are not supported. 

Can your team add to a Sprint to support disability emojis too? 
**To Reproduce**

Just try to copy/paste any major disability emoji and you'll get the 3 squares of death.
<img width="831" alt="Screen Shot 2020-02-01 at 11 00 35 AM" src="https://user-images.githubusercontent.com/19964402/73600813-a9dc5480-450a-11ea-88a6-50d8b232aadf.png">
 

**Expected behavior**

Have disability emojis show up, similar to other emojis that work ok. 

**Screenshots**

screenshot attached!

**Desktop (please complete the following information):**

- OS: macOS Mojave
- Browser: Chrome
- Version: 79.0.3945.130 

**Smartphone (please complete the following information):**

I don't use this platform on mobile so I haven't tried it there.

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: please wait until we add a `help-wanted` label to the issue before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @maestromac or @rhymes or @citizen428 from the DEV team and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by rhymes** [comment]
Hi @alequetzalli, thanks a lot for the issue, I have a question because I'm totally ignorant on the subject: where can I find more information on the emojis you were trying to insert?

Are they part of the newly released batch of emojis by the unicode consortium or something else entirely?

Basically what I'm asking is how to replicate the bug by finding out how you insert the emoji so that we can add more detail to this issue.

Thank you!

**Comment by JimmyAustin** [comment]
Dug into this out of curiosity, it looks like the first emoji is guide dog (U+1F9AE), and the second is service dog (U+1F415 U+200D U+1F9BA, which is DOG-ZERO WIDTH JOINER-SAFETYVEST). They are numbers 509 and 510 here: https://unicode.org/emoji/charts/full-emoji-list.html

They seem to have been a part of Unicode V12.

They render fine on Chrome for Windows for me, but not macOS.

**Comment by alequetzalli** [comment]
@rhymes Yes, @JimmyAustin got it! 

In macOS (what I am using), the disability guide dogs and others do not render. All you have to do is try to copy/paste the emojis that Jimmy mentioned. If you don't know how, you can even just look them up on Twitter and copy/paste from there? XD 

<img width="654" alt="Screen Shot 2020-02-02 at 7 56 38 AM" src="https://user-images.githubusercontent.com/19964402/73610930-945e3d80-4591-11ea-8e98-b77373fa01a2.png">

**Comment by rhymes** [comment]
Thank you both @JimmyAustin and @alequetzalli, I definitely noticed it doesn't render on macOS 10.14.6:

![Screenshot_2020-02-02 Full Emoji List, v13 0](https://user-images.githubusercontent.com/146201/73614644-5f89cf00-4601-11ea-98f0-d3a7013a1a50.png)

I can see those in Twitter as you screenshotted:

![Screenshot_2020-02-02 Notifications Twitter](https://user-images.githubusercontent.com/146201/73614652-6c0e2780-4601-11ea-83b2-a847d793453d.png)

One thing I noticed is that I can't find any of those in the "Emoji & Symbols" menu that every macOS app has:

![Screenshot 2020-02-02 at 9 16 49 PM](https://user-images.githubusercontent.com/146201/73614658-7f20f780-4601-11ea-83e7-0a3b698605d2.png)

I also tried by enabling "Unicode Hex input" in the keyboard system preferences:

![Screenshot 2020-02-02 at 9 26 50 PM](https://user-images.githubusercontent.com/146201/73614804-c360c780-4602-11ea-9e79-4a81b1700eda.png)

but I couldn't input emojis (it doesn't accept more than 4 chars, I read around you hare supposed to type Option + the hex code), I can only do it with the emoji & symbols window

This makes me think that macOS doesn't fully support recent versions of Unicode

I then started to investigate a bit and this is what I found:

- if I try to paste the service dog emoji in the console this is what I see:

![Screenshot 2020-02-02 at 9 34 12 PM](https://user-images.githubusercontent.com/146201/73614897-c6a88300-4603-11ea-8bd4-01a5a0331396.png)

- I tried to play with them with Python 3.8 (which has a good support for Unicode) with no avail:

```shell
>>> print("\u1F9AE")
ᾚE
>>> print("\N{POODLE}")
🐩
>>> print("\N{DOG}"
... )
🐕
>>> print("\N{DOG}")
🐕
>>> print("\N{ORANGUTAN}")
🦧
>>> print("\N{GUIDE DOG}")
🦮
>>> print("\N{SERVICE DOG}")
  File "<stdin>", line 1
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 0-14: unknown Unicode character name
>>> print("\N{WOLF}")
  File "<stdin>", line 1
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 0-7: unknown Unicode character name
>>> print("\N{FOX}")
  File "<stdin>", line 1
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 0-6: unknown Unicode character name
>>> unicodedata.lookup("CAT FACE")
'🐱'
>>> print("\N{CAT FACE}")
🐱
```

so nope, Python doesn't support all the recent emojis.

The same using the most popular emoji library:

```python
>>> emoji.emojize(":service_dog:")
'🐕\u200d🦺'
>>> emoji.emojize(":guide_dog:")
'🦮'
>>> emoji.emojize(":poodle:")
'🐩'
```

Same happens in Ruby with the newer version of the [gemoji gem](https://github.com/github/gemoji) (the one we use to interpret emojis in DEV which was developed by Github):

```ruby
[8] pry(main)> Emoji.find_by_alias("guide_dog").raw
=> "🦮"
[9] pry(main)> Emoji.find_by_alias("service_dog").raw
=> "🐕‍🦺"
[10] pry(main)> Emoji.find_by_alias("poodle").raw
=> "🐩"
```

I'm not sure how Twitter does it though but I guess it's a custom parser paired with a custom set of icons, unlike these ones which seem to rely on the rendering capabilities of the operating system like ours.

A thing we could do in the meantime is open an issue to Github's library and see if they can help somehow...

**Comment by alequetzalli** [comment]
@rhymes thank you for detailed reply! I really appreciate the level of detail you put into researching this for the disabled community. It may seem like a silly thing, but it feels nice to have an emoji that looks like me.

**Follow up question...**
Why open an issue to GitHub's library? I don't follow. O_O

**Comment by citizen428** [comment]
@alequetzalli  What @rhymes meant was that we use a library called [gemoji](https://github.com/github/gemoji) for rendering emojis on DEV. It was developed by Github, so he suggested potentially taking this feature request upstream.

**Comment by citizen428** [comment]
I did some digging myself, according to [this](https://www.apple.com/newsroom/2019/07/apple-offers-a-look-at-new-emoji-coming-to-iphone-this-fall/) Apple added Unicode 12 support last fall (iOS 13, macOS Catalina). The ones from your list I tried seem to render fine for me on Catalina:

<img width="544" alt="Screen Shot 2020-02-03 at 10 49 57" src="https://user-images.githubusercontent.com/47985/73624329-fd69b200-4672-11ea-98cc-fbd4551eb765.png">

**Comment by alequetzalli** [comment]
@citizen428 I am on macOS Mojave... perhaps Catalina is the first version to support them?!

**Comment by citizen428** [comment]
@alequetzalli Yes, the article I linked above says that Unicode 12 support was added with Catalina/iOS 13.

**Comment by rhymes** [comment]
Thanks @citizen428 for finding that out, I googled "macOS unicode versions" for a bit but I couldn't find a specific page that listed which macOS version supported which Unicode version.

I think on our part the only thing we can do is upgrade `gemoji`'s version to make sure we pick up all the new emojis and keep that updated. I'll send a PR to update this ASAP

If, as we confirmed, the limitation is due to rendering capabilities in the operating system, I don't think opening an issue to `gemoji`'s github will be of any help anyhow.

What do you think @alequetzalli @citizen428 ?

**Comment by rhymes** [comment]
Sent a PR to support the emojis server side https://github.com/thepracticaldev/dev.to/pull/5842

**Comment by alequetzalli** [comment]
@citizen428 Ah! This is so useful. So it sounds like the solution is to upgrade to Catalina... which I hear keeps crashing everyone's systems... so.. yay? 😂

**Comment by alequetzalli** [comment]
@rhymes that seems like a good solution! thank you :)
